php/class-fieldmanager-field.php

```diff
@@ -446,9 +463,29 @@ public function element_markup( $values = array() ) {
 		}
 
 		// Checks to see if element has display_if data values, and inserts the data attributes if it does
-		if ( isset( $this->display_if ) && !empty( $this->display_if ) ) {
+		if ( ! empty( $this->display_if ) ) {
 			$classes[] = 'display-if';
-			$fm_wrapper_attrs['data-display-src'] = $this->display_if['src'];
+
+			if ( isset( $this->display_if['src'] ) && isset( $this->display_if['selector'] ) ) {
+				throw new \FM_Developer_Exception(
+					esc_html( sprintf(
+						/* translators: 1: 'display-if', 2: `src`, 3: `selector` */
+						__( 'A field cannot have both a %1$s %2$s and a %1$s %3$s', 'fieldmanager' ),
+						'`$display_if`',
+						'`src`',
+						'`selector`'
```

Comment:
Do these 3 strings need to be added via `sprintf()`? Since they aren't dynamic values, is there a reason they can't just be a part of the internationalized string?

Reply:
The idea is that they're not dynamic, but they're also not translatable. I borrowed this pattern from what I've seen in some places in core, e.g. https://github.com/WordPress/WordPress/blob/master/wp-admin/network.php#L40-L49. I assume the reasoning is that it makes the remaining string easier to translate, but I can't point to any documentation about it.

Reply:
Another example, hot off the press: https://core.trac.wordpress.org/ticket/42161.

Reply:
That makes good sense, thanks for clarifying!